.pre-commit-config.yaml

```diff
@@ -1,43 +1,62 @@
 repos:
+  - repo: https://github.com/psf/black
+    rev: "21.12b0"
+    hooks:
+      - id: black
+        args:
+          - --quiet
+
+  - repo: https://github.com/pycqa/isort
+    rev: "5.10.1"
+    hooks:
+      - id: isort
+
+  - repo: https://github.com/asottile/pyupgrade
+    rev: "v2.31.0"
+    hooks:
+      - id: pyupgrade
+        args:
+          - --py3-plus
+          - --py36-plus
+          - --py37-plus
+
   - repo: https://github.com/pre-commit/pre-commit-hooks
     rev: "v4.1.0"
     hooks:
+      - id: check-ast
       - id: trailing-whitespace
       - id: end-of-file-fixer
+      - id: mixed-line-ending
+
   - repo: https://github.com/sirosen/check-jsonschema
     rev: "0.9.1"
     hooks:
       - id: check-github-workflows
         require_serial: true
+
   - repo: https://gitlab.com/pycqa/flake8
     rev: "3.9.2"
     hooks:
       - id: flake8
-  - repo: https://github.com/pre-commit/mirrors-isort
-    rev: "v5.10.1"
-    hooks:
-      - id: isort
-  - repo: https://github.com/mgedmin/check-manifest
-    rev: "0.47"
-    hooks:
-      - id: check-manifest
-  - repo: https://github.com/asottile/pyupgrade
-    rev: "v2.31.0"
-    hooks:
-      - id: pyupgrade
-        args:
-          - --py3-plus
-          - --py36-plus
-          - --py37-plus
-  - repo: https://github.com/psf/black
-    rev: "21.12b0"
-    hooks:
-      - id: black
+        additional_dependencies:
+          - flake8-type-checking
+          - flake8-bugbear
+          - flake8-comprehensions
+          - flake8-print
+          - flake8-mutable
+          - flake8-simplify
+          - flake8-pytest-style
+          - flake8-printf-formatting
+
   - repo: https://github.com/pre-commit/mirrors-mypy
     rev: "v0.930"
     hooks:
       - id: mypy
-        args: [ ]
         exclude: ^(docs/|setup.py$)
         additional_dependencies:
           - pytest
+
+  - repo: https://github.com/mgedmin/check-manifest
+    rev: "0.47"
+    hooks:
+      - id: check-manifest
```

Comment:
Moved this to the bottom, since it takes a long time to run. This way you can ctrl+c and restart the process sooner, if e.g., the black hook fails. In other words, just a proposed quality of life thing.